### ISSUE TITLE
Only *toassemble* as output. Why?

### ISSUE THREAD

**Issue by tiannagrant** [liulab-dfci/TRUST4#144]
Hello, 
I have a single cell VDJ dataset and looking to align g/d reads to IMGT mouse ref genome. This is the command I used (fastqs as input)
**

run-trust4 -f /path_mouseref/trust4/TRUST4/mouse/GRCm38_bcrtcr.fa --ref /pathto_IMGTref/trust4/TRUST4/mouse/mouse_IMGT+C.fa -u /path_to_10X_fastqs/Control_TCR_R2_.fastq.gz --barcode /path_to_10X_fastqs/Control_TCR_R1_.fastq.gz --barcodeRange 0 15 + --barcodeWhitelist /path_to/cellranger/barcodes/737K-august-2016.txt

**
after running this command Trust4 begins to " Start to extract candidate reads from read files" 
however, it stall here for hours. Not sure why? Is there an explanation? Thank you.

**Comment by mourisl** [comment]
The fastq-extractor step is usually the most time-consuming step as it needs to scan the whole fastq file. I would recommend running TRUST4 with the multiple threads with the option "-t" to speed up the process.

**Comment by tiannagrant** [comment]
I used -t but my outputs are only. I'm not sure why?
TRUST_Control_TCR_S41_L001_R2_001_toassemble.fq
TRUST_Control_TCR_S41_L001_R2_001_toassemble_bc.fa

**Comment by mourisl** [comment]
I think it is still running and is generating the candidate reads. How many reads are there in the fastq file?

**Comment by tiannagrant** [comment]
It's not longer running. There are R1:467252  & R2:742700 reads

**Comment by mourisl** [comment]
R1 is the barcode file and should have the same number of reads as the R2 file. Could you please check whether the files are intact? Thank you.

**Comment by tiannagrant** [comment]
Yes. Can you give me more specifics on how to do that? I have scdata. Should I use FASTQC or md5sum? I did confirm using md5sum that my fatsq files are exact copies from the my institutions core.

**Comment by tiannagrant** [comment]
Hm not sure what should be aware of but this is the output. 

<img width="864" alt="Screen Shot 2022-07-12 at 6 46 45 PM" src="https://user-images.githubusercontent.com/54913429/178631141-257276b7-d956-45bd-917a-c9d3fdc7753e.png">
 or 

ssm-user Control_TCR >> zcat Control_TCR_S41_L001_R1_001.fastq.gz| tail
+
FFFFFFFFFFFFFFFFFFF#FFF:FF
@A00269:636:HLKYHDSX3:1:2678:15167:37059 1:N:0:GCGTACAC
CGCCAAGCAATGACCTATTNGATCAC
+
FFFFFFFFFFFFFFFF:FF#FFFFFF
@A00269:636:HLKYHDSX3:1:2678:24008:37059 1:N:0:GCGTACAC
ACTGAGTCAGTCGATTAACNTTCACT
+
FFFFFFFFF:FFFFFFFFF#:FFFF,
ssm-user Control_TCR >> zcat Control_TCR_S41_L001_R2_001.fastq.gz| tail
+
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
@A00269:636:HLKYHDSX3:1:2678:15167:37059 2:N:0:GCGTACAC
CTAAAACCGTGAGCCTGGTGCCGGGACCGAAGTACTGTTCTCCCCCCCCCCAGTCCCCACAAGGTTACAACGTGATACAAAAAAGTTTCGT
+
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFF,F,:FFF,F:F,F,F,:,,:,,F,FFF,,FFF,:,FF:,F
@A00269:636:HLKYHDSX3:1:2678:24008:37059 2:N:0:GCGTACAC
TCAGGGACCAGGCTTTTGGTAATAGGATTGCGGCTTTCCTATGCAAGCCACCACAGCTCTCTTACATCTCAGTGTAGGAGTGAATGTGGAA
+
FFFFFF,FFFFFFF,F,FFFFFFFFFFFFFFFFFFF,FFFFFF,FF:FFFFFFFFFFFFFFFFFFF:FFFFFFF,FFFFFFFFFFFFFFF,

**Comment by mourisl** [comment]
These two files should have the same number of reads. Not sure why in your test, one is 467252 and the other is 742700. 

Another question, what TRUST4 version did you use? Thank you.

**Comment by tiannagrant** [comment]
I conda installed trust4 into my environment. Here's y output after I 'conda search -f trust4
<img width="601" alt="Screen Shot 2022-07-13 at 10 41 46 AM" src="https://user-images.githubusercontent.com/54913429/178796779-d8f4031a-d625-4f2e-9c41-ae7d56381495.png">
'

**Comment by tiannagrant** [comment]
I believe it's the latest. Thank you for helping me!

**Comment by tiannagrant** [comment]
Hi, 
I tried running trust4 again with a new command : removed the --barcodeWhitelist /path_to/cellranger/barcodes/737K-august-2016.txt option. 

OUTPUT:

system /home/ssm-user/miniconda3/envs/cyster/bin/trust4  -t 4 -f /home/ssm-user/trust4/TRUST4/mouse/GRCm38_bcrtcr.fa -o TRUST_Control_TCR_S41_L001_R2_001 -u TRUST_Control_TCR_S41_L001_R2_001_toassemble.fq --barcode TRUST_Control_TCR_S41_L001_R2_001_toassemble_bc.fa failed: 9 at /home/ssm-user/miniconda3/envs/cyster/bin/run-trust4 line 48.

Is the failed message related to unequal reads in R1 and R2?

**Comment by tiannagrant** [comment]
I should also note that I have hashtag nt in R2. Does that affect trust-4 calling process?

**Comment by mourisl** [comment]
R1 and R2 should have an equal number of reads. 
For the hashtag, you can use the "--read1Range" option to specify the range of the actual nucleotides.

**Comment by tiannagrant** [comment]
Okay, sorry this took me a while but after checking all my fastq files are the same. Also the 10X kit used to sequence my TCRs was VDJ 5' verision 1.1, paired end reads. At the moment I'm using a compute that gas 32 CPUs. How many CPUs does trust4 typically use because all are being used at the moment with this command? 
run-trust4 -t 4 -f /home/ssm-user/TRUST4/mouse/GRCm38_bcrtcr.fa --ref /home/ssm-user/TRUST4/mouse/mouse_IMGT+C.fa -1 /220412_A00269_0636_BHLKYHDSX3_fastqs/IMQ_TCR/IMQ_TCR_S42_L001_R1_001.fastq.gz --barcode /220412_A00269_0636_BHLKYHDSX3_fastqs/IMQ_TCR/IMQ_TCR_S42_L001_R1_001.fastq.gz --barcodeRange 0 15 + --barcodeWhitelist /home/ssm-user/cellranger-7.0.0/lib/python/cellranger/barcodes/737K-august-2016.txt --UMI STRING /220412_A00269_0636_BHLKYHDSX3_fastqs/IMQ_TCR/IMQ_TCR_S42_L001_R1_001.fastq.gz --umiRange 0 9 + -2 /220412_A00269_0636_BHLKYHDSX3_fastqs/IMQ_TCR/IMQ_TCR_S42_L001_R2_001.fastq.gz --read2Range 0 90
